app/assets/javascripts/discourse-markdown-it/src/options.js

```diff
@@ -0,0 +1,83 @@
+import { deepMerge } from "discourse-common/lib/object";
+
+// the options are passed here and must be explicitly allowed with
+// the const options & state below
+export default function buildOptions(state) {
```

Comment:
This was is similar to the previous `buildOptions` in `pretty-text` but it 1) no longer also call `setupIt` in one shot and 2) no longer expected to be used publicly. Instead, consumers should be using `DiscourseMarkdownIt.withOptinons` to instantiate the class directly.